compiler-core/src/format.rs

```diff
@@ -1487,22 +1500,54 @@ fn list<'a>(elems: Document<'a>, tail: Option<Document<'a>>) -> Document<'a> {
 }
 
 fn printed_comments<'a, 'comments>(
-    comments: impl IntoIterator<Item = &'comments str>,
+    comments: impl IntoIterator<Item = Option<&'comments str>>,
+    trailing_newline: bool,
 ) -> Option<Document<'a>> {
     let mut comments = comments.into_iter().peekable();
     let _ = comments.peek()?;
-    Some(concat(Itertools::intersperse(
-        comments.map(|c| "//".to_doc().append(Document::String(c.to_string()))),
-        line(),
-    )))
+
+    let mut doc = Vec::new();
+    while let Some(c) = comments.next() {
+        if let Some(c) = c {
+            doc.push("//".to_doc().append(Document::String(c.to_string())));
+            match comments.peek() {
+                // next line is a comment
+                Some(Some(_)) => doc.push(line()),
+                // next line is empty
```

Comment:
```suggestion
                // Next line is empty
```
Just me being fussy :)